.github/workflows/release.yml

```diff
@@ -45,93 +47,111 @@ jobs:
             nexusxyz/nexus-cli:${{ github.sha }}
             nexusxyz/nexus-cli:${{ github.ref_name }}
 
-
-  build-linux-x86_64:
-    name: Build Linux (x86_64)
-    runs-on: ubuntu-latest
+  build:
+    name: Build ${{ matrix.platform }}
+    runs-on: ${{ matrix.runner }}
+    strategy:
+      matrix:
+        include:
+          - platform: Linux (x86_64)
+            runner: ubuntu-latest
+            target: x86_64-unknown-linux-gnu
+            binary_name: nexus-network-linux-x86_64
+            rustflags: "-C target-feature=+crt-static"
+            build_command: cargo build --release --target x86_64-unknown-linux-gnu
+            linker: ""
+            mold: true
+          - platform: Linux (ARM64)
+            runner: ubuntu-latest
+            target: aarch64-unknown-linux-gnu
+            binary_name: nexus-network-linux-arm64
+            rustflags: "-C target-feature=+crt-static"
+            build_command: |
+              export CARGO_TARGET_AARCH64_UNKNOWN_LINUX_GNU_LINKER=aarch64-linux-gnu-gcc
+              cargo build -Zbuild-std=std,panic_abort --release --target aarch64-unknown-linux-gnu
+            linker: "aarch64-linux-gnu-gcc"
+            mold: true
+          - platform: macOS (x86_64)
+            runner: macos-latest
+            target: x86_64-apple-darwin
+            binary_name: nexus-network-macos-x86_64
+            rustflags: "-C target-feature=+sse4.2,+avx,+avx2"
+            build_command: cargo +nightly-2025-04-06 build --release --target=x86_64-apple-darwin -Z build-std=std,panic_abort
```

Comment:
Why do we only need to specify nightly here?

Reply:
I need to actually test the releases changes more. I'll split it into a separate PR